core/unit_test/TestExecutionEnvironmentNonInitializedOrFinalized.cpp

```diff
@@ -41,21 +41,83 @@ TEST_F(ExecutionEnvironmentNonInitializedOrFinalized_DeathTest,
     Kokkos::View<int> v0;
     Kokkos::View<float*> v1;
     Kokkos::View<NonTrivial**> v2;
+    return std::make_tuple(v0, v1, v2);
   };
   EXPECT_EXIT(
       {
-        make_views();
+        { (void)make_views(); }
         std::exit(EXIT_SUCCESS);
       },
       ::testing::ExitedWithCode(EXIT_SUCCESS), "");
   EXPECT_EXIT(
       {
-        Kokkos::initialize();
-        Kokkos::finalize();
-        make_views();
+        {
+          Kokkos::initialize();
+          auto views =
+              make_views();  // views outlive the Kokkos execution environment
+          Kokkos::finalize();
+        }
+        std::exit(EXIT_SUCCESS);
+      },
+      ::testing::ExitedWithCode(EXIT_SUCCESS), "");
+  EXPECT_EXIT(
+      {
+        {
+          Kokkos::initialize();
+          Kokkos::finalize();
+          (void)make_views();
+        }
         std::exit(EXIT_SUCCESS);
       },
       ::testing::ExitedWithCode(EXIT_SUCCESS), "");
 }
 
+TEST_F(ExecutionEnvironmentNonInitializedOrFinalized_DeathTest, views) {
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
+
+  EXPECT_EXIT(
+      {
+        {
+          Kokkos::View<int*> v;
+          Kokkos::initialize();
+          v = Kokkos::View<int*>("v", 10);
+          v = Kokkos::View<int*>();
```

Comment:
So, as long as a view is reassigned to a default constructed view before `finalize`, it won't trigger an abort?

Reply:
Correct.

This had been discussed at length when we were working on Kokkos 3.7 and we had agreed that we would not break this use case.  Codes like LAMMPS have been defining views as static data members.
We have just never gotten around to write test that ensure this always works (and apparently neither to document it...  I suppose some of us were not thrilled about it.)